Set up for the DB

NicholasDanks opened the pull request.

NicholasDanks committed:
Updated Gemfile

NicholasDanks committed:
Updated environments.rb

NicholasDanks committed:
Updated the migrations files - Please review

NicholasDanks committed:
Updated the Rakefile

rjollet commented on Gemfile:
```diff
@@ -4,4 +4,18 @@ ruby '2.2.3'
 gem 'thin'
 gem 'json'
+gem 'sequel'
+gem 'sqlite3'
```
sqlite3 is for developement and test but not for production so you can put 

```
group :development, :test do
  gem 'sqlite3'
end
```

NicholasDanks commented:
Done

rjollet commented on db/migrations/001_trackers_create.rb:
```diff
@@ -0,0 +1,13 @@
+  	  primary_key :id
+  	  foreign_key :campaign_id
+  	  
+  	  String :label, unique: true, null: false
```
label is not unique but the couple (campaign_id, label)

```
unique [:campaign_id, label]
```

NicholasDanks replied:
Done

rjollet commented on db/migrations/001_trackers_create.rb:
```diff
@@ -0,0 +1,13 @@
+  	  String :url, unique: true, null: false
+  	end
+  end
+end
```
you need to put an empty line at this end of your files there is this issue in all your migrations files

rjollet commented on db/migrations/002_campaign_create.rb:
```diff
@@ -0,0 +1,11 @@
+require 'sequel'
```
files name should be 002_campaigns_create.rb to be consistent

NicholasDanks committed:
Errors corrected by Renojd

rjollet commented on Gemfile:
```diff
@@ -5,12 +5,13 @@ gem 'sinatra'
   gem 'tux'
-  gem 'hirb'	
+  gem 'hirb'
+  gem 'sqlite3'	
```
you need to create a group for both test and development sqlite3 is also needed for test

```
group :development, :test do
  gem 'sqlite3'
end
```

NicholasDanks committed:
Updated models

NicholasDanks commented:
Ok guys. I completed the models. It's a little sketchy, but a good base I think. Please can you check it and agree pull.

rjollet commented on models/campaign.rb:
```diff
@@ -0,0 +1,16 @@
+  	       id: id,
+  	       label: label,
+  	  },
+  		options)
```
you mixed white-space and tab please convert tab into white-spaces

rjollet commented on models/visit.rb:
```diff
@@ -0,0 +1,22 @@
+  many_to_one :trackers
+
+  def to_json(options = {})
+  	JSON({ type: 'tracker',
```
type view

rjollet committed:
Correction of several bugs

update gitignore to not share the db/*.db
app.rb require environment and model.init
typing mistake of the class name in campaign and tracker model
delete models/pixels because we do not use this model
